### PR TITLE
Update communication channels

### LINKED ISSUE #9695: Update communication channels
Following #9690 and especially https://github.com/pypa/warehouse/pull/9690#issuecomment-863819204, we could update the communication channels advertised throughout the project:

- Wherever `distutils-sig` is mentionned, a reference to the [Discourse](https://discuss.python.org/c/packaging/14) would be more appropriate, so it should be replaced
- Wherever IRC is mentionned, adding a reference to the [PyPA Discord](https://discord.gg/pypa) is advised (but IRC should be there too, both are live)

### PR DESCRIPTION
**Pull request by meowmeowmeowcat**
Fixes #9695 

- Change `distutils-sig` to [`Discourse`](https://discuss.python.org/c/packaging/14)
- Add a reference to the PyPA Discord

### CODE REVIEW

File: docs/development/getting-started.rst [hunk 1]
```diff
@@ -652,6 +651,7 @@ Resources to help you learn Warehouse's context:
 .. _`open issues that are labelled "good first issue"`: https://github.com/pypa/warehouse/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22
 .. _`GitHub`: https://github.com/pypa/warehouse
 .. _`on Libera`: https://web.libera.chat/#pypa,#pypa-dev
+.. _`Discourse` : https://discuss.python.org/c/packaging/14
 .. _`distutils-sig mailing list`: https://mail.python.org/mailman3/lists/distutils-sig.python.org/
```
[the remark this answers is not in the document]
Comment (meowmeowmeowcat) [reply docs/development/getting-started.rst:655]: OK

[the remark this answers is not in the document]
Comment (meowmeowmeowcat) [reply docs/development/getting-started.rst:655]: It should be removed now :)

**Review by meowmeowmeowcat — COMMENTED** [review]
I've changed `discuss_href` to `discourse_href` in `help.html`.

File: warehouse/templates/pages/help.html [hunk 2]
```diff
@@ -864,11 +864,11 @@ <h3>{% trans %}Resources{% endtrans %}</h3>
 
       <h3>{% trans %}Contact{% endtrans %}</h3>
       <p>
-        {% trans trimmed pypa_href='https://pypa.io', irc_href='https://web.libera.chat/#pypa', discuss_href='https://discuss.python.org/c/packaging/14', title=gettext('External link') %}
+        {% trans trimmed pypa_href='https://pypa.io', irc_href='https://web.libera.chat/#pypa', discourse_href='https://discuss.python.org/c/packaging/14', title=gettext('External link') %}
           The <a href="{{ pypa_href }}" title="{{ title }}" target="_blank" rel="noopener">Python Packaging Authority (PyPA)</a>
           is a working group who work together to improve Python packaging. If you'd like to get in touch with a core packaging developer,
           use <a href="{{ irc_href }}" title="{{ title }}" target="_blank" rel="noopener">#pypa on IRC (Libera)</a>,
-          or <a href="{{ discuss_href }}" title="{{ title }}" target="_blank" rel="noopener">browse the online board</a>.
```
Comment (meowmeowmeowcat) [inline warehouse/templates/pages/help.html:871]: here

File: warehouse/templates/pages/help.html [hunk 3]
```diff
@@ -864,11 +864,11 @@ <h3>{% trans %}Resources{% endtrans %}</h3>
...
+          or <a href="{{ discourse_href }}" title="{{ title }}" target="_blank" rel="noopener">browse the online board</a>.
```
Comment (meowmeowmeowcat) [inline warehouse/templates/pages/help.html:871]: here

Comment (meowmeowmeowcat) [reply warehouse/templates/pages/help.html:871]: Or should we keep the same?

Comment (ewjoachim) [reply warehouse/templates/pages/help.html:871]: In this context, Discourse and discuss are the same, so whichever.

Comment (meowmeowmeowcat) [reply warehouse/templates/pages/help.html:871]: Thanks

**Review by ewjoachim — APPROVED** [review]

**Review by ewdurbin — APPROVED** [review]

**Pull request merged**